modules/tensor_mechanics/src/postprocessors/SidesetForce.C

```diff
@@ -0,0 +1,46 @@
+//* This file is part of the MOOSE framework
+//* https://www.mooseframework.org
+//*
+//* All rights reserved, see COPYRIGHT for full restrictions
+//* https://github.com/idaholab/moose/blob/master/COPYRIGHT
+//*
+//* Licensed under LGPL 2.1, please see LICENSE for details
+//* https://www.gnu.org/licenses/lgpl-2.1.html
+
+#include "SidesetForce.h"
+#include "metaphysicl/raw_type.h"
+
+registerMooseObject("TensorMechanicsApp", SidesetForce);
+registerMooseObject("TensorMechanicsApp", ADSidesetForce);
+
+template <bool is_ad>
+InputParameters
+SidesetForceTempl<is_ad>::validParams()
+{
+  InputParameters params = SideIntegralPostprocessor::validParams();
+  params.addClassDescription(
+      "This postprocessor computes the force on sidesets in parameter direction");
```

Comment:
Computes the integrated reaction force in a user-specified direction on a nodeset from the surface traction